Engine/AlgorithmManager.cs

```diff
@@ -1032,34 +1032,36 @@ private static void ProcessDelistedSymbols(IAlgorithm algorithm, List<Delisting>
             {
                 // check if we are holding position
                 var security = algorithm.Securities[delistings[i].Symbol];
-                if (security.Holdings.Quantity == 0) continue;
+                if (security.Holdings.Quantity == 0)
+                {
+                    continue;
+                }
 
                 // check if the time has come for delisting
                 var delistingTime = delistings[i].Time;
                 var nextMarketOpen = security.Exchange.Hours.GetNextMarketOpen(delistingTime, false);
                 var nextMarketClose = security.Exchange.Hours.GetNextMarketClose(nextMarketOpen, false);
 
-                if (security.LocalTime < nextMarketClose) continue;
-
-                // submit an order to liquidate on market close or exercise (for options)
-                SubmitOrderRequest request;
-
-                if (security.Type == SecurityType.Option)
+                if (security.LocalTime < nextMarketClose)
                 {
-                    // notify tx handler of the expiration, this will be handled via the configured exercise model to see if
-                    // we auto-exercise or get assigned at expiration. don't be presumptuous from here and guess if exercised/assigned
-                    request = new SubmitOrderRequest(OrderType.OptionExercise, security.Type, security.Symbol,
-                        security.Holdings.Quantity, 0, 0, algorithm.UtcTime, "Option Expired");
+                    continue;
                 }
-                else
+
+                var orderType = OrderType.Market;
+                var tag = "Liquidate from delisting";
+                if (security.Type == SecurityType.Option)
                 {
-                    request = new SubmitOrderRequest(OrderType.Market, security.Type, security.Symbol,
-                        -security.Holdings.Quantity, 0, 0, algorithm.UtcTime, "Liquidate from delisting");
+                    // tx handler will determine auto exercise/assignment
+                    tag = "Option Expired";
+                    orderType = OrderType.OptionExercise;
                 }
 
-                algorithm.Transactions.ProcessRequest(request);
+                // submit an order to liquidate on market close or exercise (for options)
+                var request = new SubmitOrderRequest(orderType, security.Type, security.Symbol,
```

Comment:
Nice to this standarized! 👍

Reply:
yea! amazing how some branches collapse on standardization!